packages/ui/src/components/Designer/RightSidebar/DetailView/index.tsx

```diff
@@ -119,6 +119,7 @@ Check this document: https://pdfme.com/docs/custom-schemas`);
         span: 12,
       },
       key: { title: i18n('fieldName'), type: 'string', required: true, span: 12 },
+      required: { title: i18n('required'), type: 'boolean', span: 6, hidden: defaultSchema?.readOnly },
```

Comment:
Ultimately, I want to eliminate the read-only schema and make it toggleable with a checkbox, similar to the Required option. 

I think it's fine to address this in a separate PR.